### PR TITLE
docs(installation): prioritize official installer and recommends nvs/nvm

### PR DESCRIPTION
**Pull request by curbengh**
## Check List
- [x] Others (Update, fix, translation, etc...)

Addresses https://github.com/hexojs/site/pull/1137#issuecomment-538716437, cc @SukkaW 

I tested nvs; the [official guide](https://github.com/jasongin/nvs) mentions installing to home folder, which means global npm packages are also installed there, so shouldn't have permission issue (I didn't experience any).

### CODE REVIEW

**Comment by SukkaW** [comment]
I have add a warning about Node.js permission in #1027 and it could be moved to `Install Node.js` part.

**Comment by curbengh** [comment]
I just moved the warning to Node.js part. I prefer this PR over #1146 :
1. nvs is much easier to install on windows compared to nodejs official's. nvs is also available on Chocolatey.
2. Mac/Linux user should use a package manager, especially for programming packages. permission issue shouldn't be a deal breaker, it's trivial to change the default global npm path.
  - Having package manager (including Chocolatey) also make it easier to update, more likely to install update (especially security update), compared to nvs/nvm.
3. nvs operates similarly to nvm on Linux (should be similar on Mac too), not necessary to recommend nvs to Window, and nvm to Mac/Linux.
4. Nodesource should be prioritized for Linux, because Ubuntu/Debian LTS still ships with Node 8. I believe Redhat also ship that version. Other Linux distro (non-LTS or rolling release) should have newer Node, so installing from default repo is fine.
5. More concise.

**Comment by SukkaW** [comment]
@curbengh The guide of installing Node.js is for anyone who has no Node.js installed. Most of exprienced Node.js developers are already using nvm, n or nvs to manage their Node.js installation.

So we should let Windows & Mac users install Node.js use official installer, and for Linux users, we can let them follows the installation guide provided by Nodesource: https://github.com/nodesource/distributions#installation-instructions

**Comment by curbengh** [comment]
Prioritized node.js installer in latest commit.

Anyway, previously I also added "[Advanced installation](https://hexo.io/docs/#Advanced-installation-and-usage)" section (as part of https://github.com/hexojs/hexo-starter/issues/3) to install hexo locally. That is also a way to avoid permission issue and I think it's the best method (compared to nvs/nvm and npm path approaches), if user is fine with changing $PATH.

Global npm package should only be installed when it's necessary to use a single version across multiple projects. This is not applicable to Hexo because most users would have only one blog folder. Installing a global hexo-cli is actually more like a workaround because it takes care of $PATH.

If anyone see linux user struggle with permission issue, I recommend to point to "[Advanced installation](https://hexo.io/docs/#Advanced-installation-and-usage)" section. Or `$ echo "prefix=$HOME/.local" >> ~/.npmrc`.

I also intend to provide Snap and Flatpak packages as another alternative for Linux user, though Nodejs binary still needs to be installed beforehand.

File: source/docs/index.md [hunk 1]
```diff
@@ -24,29 +24,31 @@ If your computer already has these, congratulations! You can skip to the [Hexo i
...
+
+- Windows: Install it with [nvs](https://github.com/jasongin/nvs/) (recommended), [nvm](https://github.com/nvm-sh/nvm) or [Chocolatey](https://chocolatey.org/). (nvs is also recommended for Mac and Linux to avoid possible permission issue)
 - Mac: Install it with [Homebrew](https://brew.sh/) or [MacPorts](http://www.macports.org/).
 - Linux (DEB/RPM-based): Install it with [NodeSource](https://github.com/nodesource/distributions).
-- Linux (other): Install it through respective package manager (if Node.js >= 10.0 is available) or use the installation scripts mentioned in Windows.
+- Linux (other): Install it through respective package manager (if Node.js >= 10.0 is available) or use nvs/nvm.
 
-Alternatively, download and run the [official installer](https://nodejs.org/en/download/) provided by Node.js.
+{% note warn Mac / Linux %}
+If you encounter permission error when trying to install Hexo, please follow the workaround [provided by npmjs](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally); overriding with root/sudo is highly discouraged.
```
Comment (SukkaW) [inline source/docs/index.md]:
```suggestion
If you encounter `EACCES` error when when trying to install Hexo. Please follow [the workaround provided by npmjs](https://docs.npmjs.com/resolving-eacces-permissions-errors-when-installing-packages-globally); overriding with root/sudo is highly discouraged.
```

Also this notice might need to be placed under `Install Hexo` section.

File: source/docs/index.md [hunk 2]
```diff
@@ -24,29 +24,31 @@ If your computer already has these, congratulations! You can skip to the [Hexo i
...
+{% endnote %}
+
 ### Install Node.js
 
-- Windows: Install it with [nvs](https://github.com/jasongin/nvs/) (recommended) or [nvm](https://github.com/nvm-sh/nvm). (Either also support Mac and Linux)
+Node.js provides [official installer](https://nodejs.org/en/download/) for most platforms.
+
+Alternative installation methods:
+
+- Windows: Install it with [nvs](https://github.com/jasongin/nvs/) (recommended), [nvm](https://github.com/nvm-sh/nvm) or [Chocolatey](https://chocolatey.org/). (nvs is also recommended for Mac and Linux to avoid possible permission issue)
```
Comment (SukkaW) [inline source/docs/index.md]:
```suggestion
- Windows: Install it with [nvs](https://github.com/jasongin/nvs/) (recommended), [nvm](https://github.com/nvm-sh/nvm) or [Chocolatey](https://chocolatey.org/).
```

File: source/docs/index.md [hunk 3]
```diff
@@ -24,29 +24,31 @@ If your computer already has these, congratulations! You can skip to the [Hexo i
...
 - Mac: Install it with [Homebrew](https://brew.sh/) or [MacPorts](http://www.macports.org/).
 - Linux (DEB/RPM-based): Install it with [NodeSource](https://github.com/nodesource/distributions).
-- Linux (other): Install it through respective package manager (if Node.js >= 10.0 is available) or use the installation scripts mentioned in Windows.
+- Linux (other): Install it through respective package manager (if Node.js >= 10.0 is available) or use nvs/nvm.
```
Comment (SukkaW) [inline source/docs/index.md]:
```suggestion
- Linux (other): Install it through respective package manager (if Node.js >= 10.0 is available).

[nvs](https://github.com/jasongin/nvs/) and [nvm](https://github.com/nvm-sh/nvm) are also recommended for Mac and Linux.
```

The recommendation for Mac & Linux should be here.

File: source/docs/index.md [hunk 4]
```diff
@@ -24,29 +24,31 @@ If your computer already has these, congratulations! You can skip to the [Hexo i
...
 - Linux (Fedora, Red Hat, CentOS): `sudo yum install git-core`
 
+{% note warn For Mac users %}
+You may encounter some problems when compiling. Please install Xcode from App Store first. After Xcode is installed, open Xcode and go to **Preferences -> Download -> Command Line Tools -> Install** to install command line tools.
+{% endnote %}
+
 ### Install Node.js
 
-- Windows: Install it with [nvs](https://github.com/jasongin/nvs/) (recommended) or [nvm](https://github.com/nvm-sh/nvm). (Either also support Mac and Linux)
+Node.js provides [official installer](https://nodejs.org/en/download/) for most platforms.
```
Comment (SukkaW) [inline source/docs/index.md]:
```
{% note info For Windows users %}
When install Node.js using official installer, make sure **Add to PATH** is checked.
{% endnote %}
```

An `Add to PATH` notice for Windows.

**Comment by SukkaW** [comment]
Node.js has an "official guide" for installing Node.js through package manager: https://nodejs.org/en/download/package-manager/

**Comment by curbengh** [comment]
> Node.js has an "official guide" for installing Node.js through package manager: https://nodejs.org/en/download/package-manager/

Thanks. Added the link. I still retain the Mac/Linux package manager list, otherwise the guide looks too Windows-ish.

**Pull request merged**